boards/hail/src/main.rs

```diff
@@ -169,14 +169,12 @@ unsafe fn set_pin_primary_functions(peripherals: &Sam4lDefaultPeripherals) {
     peripherals.pb[15].configure(None); //... D1
 }
 
-/// Reset Handler.
+/// Board's main function.
 ///
-/// This symbol is loaded into vector table by the SAM4L chip crate.
-/// When the chip first powers on or later does a hard reset, after the core
-/// initializes all the hardware, the address of this function is loaded and
-/// execution begins here.
+/// This is called from the reset handler after memory initialization is
+/// complete.
 #[no_mangle]
-pub unsafe fn reset_handler() {
+pub unsafe fn main() {
```

Comment:
```suggestion
pub unsafe fn rust_entry_point() {
```

Maybe this is a more direct name. `main` is a bit overloaded in what it means, but I believe the term "entry point" to be fairly standardized as the 'first code that is run', so calling this the "rust entry point" is perhaps a more clear term for folks reading this?

Reply:
I thought main was pretty clear, since it's where all of the interesting work starts but is not the _first_ thing that runs.

`rust_entry_point` is more grepable, however.